packages/aws-cdk-lib/aws-dynamodb/lib/table-v2.ts

```diff
@@ -516,6 +533,7 @@ export class TableV2 extends TableBaseV2 {
     this.partitionKey = props.partitionKey;
     this.hasSortKey = props.sortKey !== undefined;
     this.region = this.stack.region;
+    this.tags = new TagManager(TagType.STANDARD, CfnGlobalTable.CFN_RESOURCE_TYPE_NAME, props.tags);
```

Comment:
We don't typically add TagManager to the L2 resource. Typically the Cfn L1 resource is already extending ITaggable and the tags are directly added to the L1 resource. This might be some edge case we need to fix in the L1 generation.

Reply:
I took a deep dive into the Global Table usage and you're right. We wouldn't be able to make L1 ITaggable directly. I agree with you that this is a good workaround. 

My second question is on why did you choose to use TagManager (which is the TagV2 solution) instead of directly using the V1. I believe using V1, you can directly add 

```
protected readonly tags?: CfnTag[];
```

Reply:
You didn't make the construct `ITaggable` or `ITaggableV2` either right? When you use `Tags.of().add`, it is essentially adding an Aspect that will run during synthesis, https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/lib/tag-aspect.ts#L74. It will subsequently check whether if it's taggable using V1 or V2. 

For V1 tagging, it will only check the following:
```
const tags = (construct as any).tags;
return tags !== undefined && tags !== null && typeof tags === 'object' && (tags as any)[TAG_MANAGER_SYM];
```

I'm not sure when should we use V1 and when to use V2, i.e. when to define a tags object and when to define tagManager.

Reply:
Ahh nice catch, I didn't realize `ITaggable` interface uses `TagManager`.